src/impl_methods.rs

```diff
@@ -2049,6 +2050,41 @@ where
         self.zip_mut_with(rhs, |x, y| *x = y.clone());
     }
 
+    /// Perform an elementwise assigment of values from `self` into array or producer `to`.
+    ///
+    /// `to` can be another array or a producer of assignable elements.
+    /// [`AssignElem`] determines how elements are assigned.
+    ///
+    /// **Panics** if shapes disagree.
+    pub fn assign_into<P>(&self, to: P)
+    where
+        S: Data,
+        P: IntoNdProducer<Dim = D>,
+        P::Item: AssignElem<A>,
+        A: Clone,
+    {
+        self.map_assign_into(A::clone, to)
+    }
+
+    /// Perform an elementwise assigment of values mapped with `f` from `self` into array or
+    /// producer `to`.
+    ///
+    /// `to` can be another array or a producer of assignable elements.
+    /// [`AssignElem`] determines how elements are assigned.
+    ///
+    /// **Panics** if shapes disagree.
+    pub fn map_assign_into<'a, F, B, P>(&'a self, f: F, to: P)
```

Comment:
Maybe skip exposing this as a public method, it's not as used.

Reply:
Yeah, I'm not sure this method is necessary. The parameters are also a little awkward, because while I think the order `f: F, to: P` better fits the name of the method, it's more common for Rust methods to put the closure as the last parameter since closures are often formatted as multiple lines. (Consider e.g. `Result::map_or`, which IMO would better fit the name as `map_or<U, F>(self, f: F, default: U)` but is actually `map_or<U, F>(self, default: U, f: F)` in the standard library.)